### ISSUE TITLE
upgrading from 2.4.3 to 2.5 with docker container

### ISSUE THREAD

**Issue by paolofrs** [thingsboard/thingsboard#2779]
I cannot upgrade from older versions of thingsboard/tb-postgres image. I am following this [guide](https://thingsboard.io/docs/user-guide/install/docker/?ubuntuThingsboardQueue=inmemory) but when I run `docker run -it -v ~/.mytb-data:/data --rm thingsboard/tb-postgres upgrade-tb.sh` I get the following error:

    pg_ctl: another server might be running; trying to start server anyway    
    waiting for server to start.... stopped waiting                                                                                                                        
    pg_ctl: could not start server                                                                                                                                        
    Examine the log output.                                                                                                                                                      
    Starting ThingsBoard upgrade ...
    ===================================================
    :: ThingsBoard ::       (v2.5.0)
    ===================================================
    Starting ThingsBoard Upgrade from version 2.4.2 ...                                                                                                                     
    Upgrading ThingsBoard from version 2.4.2 to 2.4.3 ...                                                                                                                   
    Unexpected error during ThingsBoard installation!                                                                                                                       
    org.postgresql.util.PSQLException: Connection to localhost:5432 refused. Check that the hostname 
    and port are correct and that the postmaster is accepting TCP/IP connections.

Then a lot of java errors and at the end:

    Unexpected error during ThingsBoard installation!                                                                                                                       
    pg_ctl: could not send stop signal (PID: 11): No such process

I've checked and I have not the `pg_ctl.pid` file in the `<mytb_folder>/data/db/.` folder I do not know if this help. I am able to upgrade from 2.4.2 using `thingsboard/tb-postgres:2.4.3` image. Moreover, using the 2.4.3 image I am also able to run tb-postgres container while 2.5 do not work, it complains about postgresql database (and pg_ctl).

**Comment by Akjo4712** [comment]
Same Issue on my Docker Installation. 

Output in `/var/log/postgres/postgres.log` upon starting Docker with `upgrade-tb.sh`.
```
2020-05-18 06:40:03.806 UTC [14] FATAL:  database files are incompatible with server
2020-05-18 06:40:03.806 UTC [14] DETAIL:  The data directory was initialized by PostgreSQL version 9.6, which is not compatible with this version 11.7 (Debian 11.7-2.pgdg90+1).
```

**Comment by paolofrs** [comment]
seems that we have to upgrade PostgreSQL first (this page [here](https://thingsboard.io/docs/user-guide/install/upgrade-instructions/#upgrading-to-25) might help). Your log seems to confirm that. [This](https://gist.github.com/ShvaykaD/1f0e6c1321a0a2b4b9f3b9ea9ab3e8d3) is a guide to update PostgreSQL on Ubuntu (taken from the previous link). 

Any idea how to do it in thingsboard container/docker?

**Comment by Akjo4712** [comment]
Yes, you are right, this seems to be the problem.  

I think Upgrading the Database to the new Version should be part of the skript `upgrade-tb.sh` which is launched during upgrade, not something the end users should do on theire own, as this might lead to many problems with inexpirenced end users.  
At least there should be an official advice, how to do this in a supported way.

**Comment by ashvayka** [comment]
Hi @paolofrs There is not simple 1 line solution here. The previous version of the image was using PostgreSQL 9.6 and now we require version 11. I suggest to copy the database files from ~/.mytb-data and update them. Then you should be able to launch a separate docker container for postgresql 11 (using that data folder) and separate for ThingsBoard. This can be done using Docker Compose scripts: https://github.com/thingsboard/thingsboard/tree/master/docker

**Comment by mpolitze** [comment]
Hi @paolofrs, I had the same issue when upgrading from v2.4.3 to 3.0.1 and maybe someone else stumbles upon this during the upgrade. The docker containers [here](https://github.com/tianon/docker-postgres-upgrade) greatly helped me to run the upgrade "the docker way". I made a backup and then ran:
```sh
docker run --rm -v /var/thingsboard2/data/db:/var/lib/postgresql/9.6/data -v /var/thingsboard3/data/db:/var/lib/postgresql/11/data --env LANG=C.UTF-8 tianon/postgres-upgrade:9.6-to-11
```
I had to add the `LANG` env var to match the `lc_collate` from the `tb-postgress` docker image to the sample call of the docker image. Then I copied over the rest of my `thingsboard2` dir, fixed ownners, and ran the usual  upgrade procedure in the new data directory.

**Comment by paolofrs** [comment]
hi @mpolitze , many tks for your help! Your solution works pretty much out of the box. I've just struggled a bit with the owners. On the `~/mytb-data/db` one should use `systemd-resolve:kvm` while `799:799` for all the rest (at lest it works for me with this configuration). 

Another problem resides in the `thingsboard.yml` configuration file. It differs from different release, thus I had to start the container without passing an external configuration file, copy the automatically generated from within the container to local file-system, edit it and import my configuration from old file and re-add it in the compose file.
